doc/rtd/topics/instancedata.rst

```diff
@@ -509,14 +509,19 @@ EC2 instance:
 Using instance-data
 ===================
 
-As of cloud-init v. 18.4, any variables present in
-``/run/cloud-init/instance-data.json`` can be used in:
+As of cloud-init v. 18.4, any instance-data can be used in:
 
 * User-data scripts
 * Cloud config data
 * Command line interface via **cloud-init query** or
   **cloud-init devel render**
 
+This means that any variable present in
```

Comment:
+1 updating docs here!

Here's another supplement that hopefully we add a bit of clarity for the original bug-filer as well

```diff
diff --git a/doc/rtd/topics/instancedata.rst b/doc/rtd/topics/instancedata.rst
index 2edf8d2a..8f363f48 100644
--- a/doc/rtd/topics/instancedata.rst
+++ b/doc/rtd/topics/instancedata.rst
@@ -564,9 +564,36 @@ Below are some examples of providing these types of user-data:
    {%- endif %}
    ...
 
+
+One way to easily explore what Jinja variables are available on your machine
+is to use the cloud-init query --format (-f) commandline option which will
+render any Jinja syntax you use. Warnings or exceptions will be raised on
+invalid instance-data keys, paths or invalid syntax.
+
+.. code-block:: shell-session
+ # List all instance-data keys and values as root user
+ sudo cloud-init query --all
+
+ # Introspect nested keys on an object
+ cloud-init query -f "{{ds.keys()}}"
+ dict_keys(['meta_data', '_doc'])
+
+ # Test your Jinja rendering syntax on the command-line directly
+ cloud-init query -f "{{invalid.instance-data.key}}"
+ WARNING: Ignoring jinja template for query commandline: 'invalid' is undefined
+
+ # Introspect available keys on an object in jinja
+ cloud-init query -f "{{v1.keys()}}"
+ WARNING: Could not render jinja template variables in file 'query commandline': 'invalidkey'
+ CI_MISSING_JINJA_VAR/invalidkey
+
+ # Substitute available nested keys from instance-data
+ cloud-init query -f "{{ds.meta_data.ami_id}}"
+
+
 .. note::
   Trying to reference jinja variables that don't exist in
-  instance-data.json will result in warnings in ``/var/log/cloud-init.log``
+  instance-data will result in warnings in ``/var/log/cloud-init.log``
   and the following string in your rendered user-data:
   ``CI_MISSING_JINJA_VAR/<your_varname>``.
 
```

Reply:
@blackboxsw , thanks that'll be a good addition. I'm not sure exactly what you're trying to show with your bottom two examples though. Particularly these two:
```
+ # Introspect available keys on an object in jinja
+ cloud-init query -f "{{v1.keys()}}"
+ WARNING: Could not render jinja template variables in file 'query commandline': 'invalidkey'
+ CI_MISSING_JINJA_VAR/invalidkey
+
+ # Substitute available nested keys from instance-data
+ cloud-init query -f "{{ds.meta_data.ami_id}}"
```

Did you lose the output from the bottom example?

Reply:
Oops I did. Updated and corrected those docs. thanks @TheRealFalcon